pkg/kubeapiserver/authenticator/config.go

```diff
@@ -55,7 +55,6 @@ import (
 
 // Config contains the data on how to authenticate a request to the Kube API Server
 type Config struct {
-	Anonymous      bool
```

Comment:
after seeing what you had to do to default the config from flags-or-file, and preserve the defaulted value by mutating the reloaded config file, I think you were closer on this line with the first version I reviewed... sorry :-/

I pushed a commit on top of the current state to https://github.com/vinayakankugoyal/kubernetes/commits/kep4633/ if you want to pull that in that makes the loading / mutual exclusive behavior / verification of immutability on config file reload all work like I'd expect

If you want to squash that in, I can take a final pass

Reply:
Done.